cgsmiles/tests/test_write_cgsmiles.py

```diff
@@ -42,5 +46,27 @@ def test_write_mol_graphs(input_string):
     mol_graph = read_cgsmiles(input_string)
     out_string = write_cgsmiles_graph(mol_graph)
     out_graph = read_cgsmiles(out_string)
-    print(out_string)
     assertEqualGraphs(mol_graph, out_graph)
+
+@pytest.mark.parametrize('input_string',(
+                        # smiple linear seqeunce
+                        "{[#PEO][#PMMA][#PEO][#PMMA]}.{#PEO=[>]COC[<],#PMMA=[>]CC(C)[<]C(=O)OC}",
+                        # something with ring
+                        "{[#TC5]1[#TC5][#TC5]1}.{#TC5=[$]cc[$]}",))
+def test_write_cgsmiles(input_string):
+    resolver = MoleculeResolver.from_string(input_string)
+    fragment_dicts = resolver.fragment_dicts
+    molecule = resolver.molecule
+    output_string = write_cgsmiles(molecule, fragment_dicts)
+    out_resolver =  MoleculeResolver.from_string(output_string)
+    out_mol = out_resolver.molecule
+    assertEqualGraphs(molecule, out_mol)
+    out_fragments = out_resolver.fragment_dicts
+    for frag_dict, frag_dict_out in zip(fragment_dicts, out_fragments):
+        assert set(frag_dict_out) == set(frag_dict)
+        for fragname in frag_dict:
+            # we cannot be sure that the atomnames are the same because they
+            # will depend on the order
+            nx.set_node_attributes(frag_dict_out[fragname], None, "atomname")
+            nx.set_node_attributes(frag_dict[fragname], None, "atomname")
```

Comment:
Can't we assume anything? For example that the first (two?) character should be the same?

Reply:
It is just the element plus the atom index. The element is already part of the testing. Therefore, I think it is not required.